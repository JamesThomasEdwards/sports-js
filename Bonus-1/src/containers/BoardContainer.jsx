// React;
import React from 'react';
import { Component } from 'react';
// Files;
import Counter from '../components/Counter.jsx';
import Button from '../components/Button.jsx';

export default class BoardContainer extends Component {
    state = {
        score: 0
    };
    increaseScore = () => {
        this.setState(() => ({ score: this.state.score + 1 }));
    };
    decreaseScore = () => {
        this.setState(() => ({ score: this.state.score - 1 }));
    };
    resetScore = () => {
        this.setState(() => ({ score: 0 }))
    }
    render() {
        const buttonHandelers = [this.increaseScore, this.decreaseScore, this.resetScore]
        const buttonTitles = ['add', 'subtract', 'reset']
        const buttons = buttonHandelers.map((ele, ind) => {
            return <Button id={ind} handeler={ele}
                buttonTitles={buttonTitles} />
        })
        return (
            <div>
                <Counter score={this.state.score} />
                {buttons}
            </div>
        )
    }
}
