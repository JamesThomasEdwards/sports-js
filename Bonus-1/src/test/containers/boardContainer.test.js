import React from 'react';
import { shallow } from 'enzyme';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import BoardContainer from '../../containers/BoardContainer.jsx';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


configure({ adapter: new Adapter() });

test('should render BoardContainer component correctly', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<BoardContainer />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

describe('BoardContainer test', () => {
    describe('<BoardContainer />', () => {
        const wrapper = shallow(<BoardContainer startValue={0} />);
        it('Renders a <div>', () => {
            expect(wrapper.type()).toEqual('div');
        });
    });
});
