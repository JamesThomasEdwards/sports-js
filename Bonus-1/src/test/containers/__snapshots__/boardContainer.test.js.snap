// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BoardContainer component correctly 1`] = `
<div>
  <Counter />
  <Button
    buttonTitles="ADD (+)"
    handeler={[Function]}
    id={0}
  />
  <Button
    buttonTitles="SUBTRACT (-)"
    handeler={[Function]}
    id={1}
  />
  <Button
    buttonTitles="RESET"
    handeler={[Function]}
    id={2}
  />
</div>
`;
